### PR TITLE
[Docs] Added static site generation for docs using Hugo

### PR DESCRIPTION
**Pull request by ghost**
Added Hugo for static site generation.
Added Dockerfile for Hugo + Apache
Added apache.conf to direct to docs.slapbirdapm.com **<- needs review**
Added docker-compose 

Currently using the hugo-dead-simple theme from https://themes.gohugo.io
Can be changed.

Only displays the getting started info from slapbirdapm.com/getting-started

### CODE REVIEW

**Comment by rawleyfowler** [comment]
LGTM so far, please however run `httpd` on another port, say `5555`, instead of `80`.

**Comment by rawleyfowler** [comment]
We'll hold on merging this until we have some more docs to add as well.

**Comment by ghost** [comment]
Will update httpd port to 5555.
Any preferences on the theme/layout etc?

**Comment by rawleyfowler** [comment]
I like the book theme, we could probably roll with that.

**Comment by rawleyfowler** [comment]
Fixes #6

**Comment by ghost** [comment]
Latest commit contains the port and theme changes, documentation for all plugins as well as for the arguments.

File: docs/content/arguments.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+{{< tab "Dancer2" >}}
+Set the ignored_headers in your config.yaml.
+```
+set plugins => {
+    SlapbirdAPM => {
+        ignored_headers => ['Example-Header']
+    }
+};
+```
```
Comment (rawleyfowler) [inline docs/content/arguments.md]: Dancer2 uses a `config.yml` file.

This should really be:

```yaml
plugins:
    SlapbirdAPM:
        ignored_headers:
          - 'Header'
```

File: docs/content/arguments.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+{{< tab "Dancer2" >}}
+Set the API key in your config.yaml.
+```
+set plugins => {
+    SlapbirdAPM => {
+        key => 'your_slapbirdapm_api_key'
+    }
+}
+```
+{{< /tab >}}
```
Comment (rawleyfowler) [inline docs/content/arguments.md]: Should be the YAML equivalent also:

```yaml
plugins:
    SlapbirdAPM:
        key: 'your_slapbirdapm_api_key'
```

File: docs/content/arguments.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+{{< tab "Dancer2" >}}
+Enable or disable topology in your config.yaml.
+```
+# disable topology
+set plugins => {
+    SlapbirdAPM => {
+        topology => 0
+    }
+}
+```
```
Comment (rawleyfowler) [inline docs/content/arguments.md]: Should be YAML aswell

**Review by rawleyfowler — COMMENTED** [review]
The `public/` directory should be `.gitignored` (MAKE SURE YOU JUST GIT IGNORE THIS ONE AND NOT THE ROOT PUBLIC DIRECTORY). We'll generate this on the server at build-time.

File: docs/content/arguments.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+{{< tab "Dancer2" >}}
+Enable or disable quiet mode in your config.yaml.
+```
+set plugins => {
+    SlapbirdAPM => {
+        quiet => 1
+    }
+}
+```
```
Comment (rawleyfowler) [inline docs/content/arguments.md]: Should be YAML as well

File: docs/content/arguments.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+{{< tab "Dancer2" >}}
+Enable or disable stacktraces in your config.yaml.
+```
+# disbale stacktraces
+set plugins => {
+    SlapbirdAPM => {
+        trace => 0
+    }
+}
+```
```
Comment (rawleyfowler) [inline docs/content/arguments.md]: Typo, plus should be yaml

File: docs/content/arguments.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+## Trace Modules
+Set modules to be be traced.
+{{< tabs "traceM_args" >}}
+{{< tab "Mojo" >}}
+Pass the modules to be traced as part of the plugin setup.
+```
+use Mojolicious::Lite -signatures;
+
+plugin 'SlapbirdAPM',
+    trace_modules => ['Example_Module'];
```
Comment (rawleyfowler) [inline docs/content/arguments.md]: A more "Perl" way to write an example module name is `Example::Module` or `My::Module`.

**Comment by rawleyfowler** [comment]
LGTM

**Pull request merged**